### ISSUE TITLE
 Performance issue in the definition of __main__, examples/circular_ode_demo.py(P1)

### ISSUE THREAD

**Issue by DLPerf** [titu1994/tfdiffeq#13]
Hello, I found a performance issue in the definition of `__main__`, examples/circular_ode_demo.py, [`tf.reduce_mean(tf.abs(pred_y - true_y))`](https://github.com/titu1994/tfdiffeq/blob/ef646f85cbd0821749a03e7ab51e03e16798fab1/examples/circular_ode_demo.py#L186) will be calculated repeatedly during program execution, resulting in reduced efficiency. I think `pred_y` in line 185 and `loss` should be created before the loop.

Looking forward to your reply. Btw, I am very glad to create a PR to fix it if you are too busy.
